x-pack/plugin/esql/src/main/java/generated/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvAvgIntEvaluator.java

```diff
@@ -35,117 +35,109 @@ public String name() {
    * Evaluate blocks containing at least one multivalued field.
    */
   @Override
-  public Block.Ref evalNullable(Block.Ref ref) {
-    try (ref) {
-      IntBlock v = (IntBlock) ref.block();
-      int positionCount = v.getPositionCount();
-      try (DoubleBlock.Builder builder = driverContext.blockFactory().newDoubleBlockBuilder(positionCount)) {
-        CompensatedSum work = new CompensatedSum();
-        for (int p = 0; p < positionCount; p++) {
-          int valueCount = v.getValueCount(p);
-          if (valueCount == 0) {
-            builder.appendNull();
-            continue;
-          }
-          int first = v.getFirstValueIndex(p);
-          if (valueCount == 1) {
-            int value = v.getInt(first);
-            double result = MvAvg.single(value);
-            builder.appendDouble(result);
-            continue;
-          }
-          int end = first + valueCount;
-          for (int i = first; i < end; i++) {
-            int value = v.getInt(i);
-            MvAvg.process(work, value);
-          }
-          double result = MvAvg.finish(work, valueCount);
+  public Block evalNullable(Block fieldVal) {
```

Comment:
Similarly here (and the other `eval...` functions inherited): should we update the javadoc for the super, abstract function? The contract is that the return value is an owned block, now, which must be closed by the caller.

Reply:
see https://github.com/elastic/elasticsearch/pull/102513/commits/d6d79023a4b934b937b879008a585e775692e944